packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -134,8 +135,16 @@ class MessagePayload {
     }
 
     const enforce_nonce = Boolean(this.options.enforceNonce);
-    if (enforce_nonce && nonce === undefined) {
-      throw new DiscordjsError(ErrorCodes.MessageNonceRequired);
+
+    // If `nonce` isn't provided, generate one & set `enforceNonce`
+    // Unless `enforceNonce` is explicitly set to `false`(not just falsy)
+    if (nonce === undefined) {
+      if (this.options.enforceNonce !== false && this.client.options.enforceNonce) {
+        this.options.nonce = DiscordSnowflake.generate().toString();
+        this.options.enforceNonce = true;
```

Comment:
Changing just `this.options` is useless since later at line 223-224 the variables above are re-used. You'll wanna alter the variables (`nonce =` from line 129 instead of `this.options.nonce` and `enforce_nonce` above